### PR TITLE
Return 400, not 412, for invalid rule PUTs

### PR DESCRIPTION
**Pull request by esnible**
For #707

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/711)

### CODE REVIEW

File: pkg/config/apiserver.go [hunk 1]
```diff
@@ -333,10 +333,10 @@ func httpStatusToRPC(httpStatus int) (code rpc.Code) {
...
-	http.StatusConflict:           rpc.ALREADY_EXISTS,
-	http.StatusForbidden:          rpc.PERMISSION_DENIED,
-	http.StatusUnauthorized:       rpc.UNAUTHENTICATED,
-	http.StatusPreconditionFailed: rpc.FAILED_PRECONDITION,
+	http.StatusOK:           rpc.OK,
+	http.StatusNotFound:     rpc.NOT_FOUND,
+	http.StatusConflict:     rpc.ALREADY_EXISTS,
+	http.StatusForbidden:    rpc.PERMISSION_DENIED,
+	http.StatusUnauthorized: rpc.UNAUTHENTICATED,
+	http.StatusBadRequest:   rpc.FAILED_PRECONDITION,
```
Comment (mandarjog) [inline pkg/config/apiserver.go]: https://github.com/grpc-ecosystem/grpc-gateway/blob/master/runtime/errors.go#L23

Please update the rpc.Code mapping per the above file.

Comment (douglas-reid) [inline pkg/config/apiserver.go]: I would have expected us to update the code to generate rpc.INVALID_ARGUMENT and to add a mapping here to http.StatusBadRequest (as opposed to change this mapping).

**Comment by esnible** [comment]
Is there anything I should do to move this PR along?  I have addressed the initial review.

**Comment by esnible** [comment]
jenkins rebuild mixer/e2e-smoketest

**Comment by esnible** [comment]
jenkins rebuild mixer/manager-regression

**Pull request merged**